payments_service/braintree/views.py

```diff
@@ -1,6 +1,97 @@
+import logging
+
+from django.core.exceptions import ObjectDoesNotExist
+
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from slumber.exceptions import HttpClientError
+
+from .. import solitude
+from ..base.views import error_400
 from ..solitude import SolitudeBodyguard
+from .forms import SubscriptionForm
+
+log = logging.getLogger(__name__)
 
 
 class TokenGenerator(SolitudeBodyguard):
+    """
+    Generate a client token to begin processing payments.
+    """
     methods = ['post']
     resource = 'braintree.token.generate'
+
+
+class Subscribe(APIView):
+    """
+    Subscribe the logged in buyer to a plan.
+    """
+    def post(self, request):
+        form = SubscriptionForm(request.DATA)
+        if not form.is_valid():
+            return error_400(response=form.errors)
+
+        buyer_uuid = request.user.pk
+        api = solitude.api()
+        try:
+            self.set_up_customer(buyer_uuid)
+            pay_method_uri = self.get_pay_method(
+                buyer_uuid,
+                form.cleaned_data['pay_method_uri'],
+                form.cleaned_data['pay_method_nonce'])
+
+            api.braintree.subscription.post({
+                'paymethod': pay_method_uri,
+                'plan': form.cleaned_data['plan_id'],
+            })
+        except HttpClientError, exc:
+            return error_400(exception=exc)
```

Comment:
I don't think in all cases that 400s should be caught and returned. In some cases an error might indicate a programming error in payments-service, not something related to user input. I'll have to think about it more though.